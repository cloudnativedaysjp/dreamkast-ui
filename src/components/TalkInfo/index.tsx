export { TalkInfo } from './TalkInfo'
export { TalkInfoLT } from './TalkInfoLT'
