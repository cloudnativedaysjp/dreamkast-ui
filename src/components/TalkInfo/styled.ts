import styled from 'styled-components'

export const Container = styled.div`
  height: 490px;
  background: rgba(255, 255, 255, 0.7);
  padding: 10px;
  border-radius: 10px;
  overflow: scroll;
  &::-webkit-scrollbar {
    display: none;
  }
`

export const OuterContainer = styled.div`
  padding: 0 0 43px 0;
`

export const Title = styled.h2`
  color: #037f8c;
`

export const Content = styled.div`
  padding: 10px 0;
`
