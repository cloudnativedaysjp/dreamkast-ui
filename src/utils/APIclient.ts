import fetch from 'node-fetch'
import { Event, Track, Talk } from '../interfaces'

/***
  Event API
***/
type GetEventParam = { eventId: number }
export async function getEvent(param: GetEventParam): Promise<Event> {
  const res = await fetch(`${process.env.API_URL}/events/${param.eventId}`)
  return (await res.json()) as Event
}

/***
  Track API
***/
type GetAllTracksParam = { eventId: number }
export async function getTracks(param: GetAllTracksParam): Promise<Track[]> {
  const res = await fetch(`${process.env.API_URL}/tracks/${param.eventId}`)
  return (await res.json()) as Track[]
}

type GetTrackParam = { trackId: number }
export async function getTrack(param: GetTrackParam): Promise<Track> {
  const res = await fetch(`${process.env.API_URL}/tracks/${param.trackId}`)
  return (await res.json()) as Track
}

/***
  Talk API
***/
type GetTalksParam = { eventId: number; trackId?: number }
export async function getTalks(param: GetTalksParam): Promise<Talk[]> {
  const URL = param.trackId
    ? `${process.env.API_URL}/talks/${param.eventId}?trackId=${param.trackId}`
    : `${process.env.API_URL}/talks/${param.eventId}`
  const res = await fetch(URL)
  return (await res.json()) as Talk[]
}

type GetTalkParam = { talkId: number }
export async function getTalk(param: GetTalkParam): Promise<Talk> {
  const res = await fetch(`${process.env.API_URL}/talks/${param.talkId}`)
  return (await res.json()) as Talk
}
