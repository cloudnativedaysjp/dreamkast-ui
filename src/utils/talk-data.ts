import { Talk } from '../interfaces'

export const Talks: Talk[] = [
  {
    id: 1,
    trackId: '1',
    vimeoId: '453943528',
    title: 'ほげふが',
    description: 'ぴよぴよ',
    speakers: ['jacopen1', 'r_takaishi2'],
  },
  {
    id: 2,
    trackId: '1',
    vimeoId: '450845161',
    title: 'ふー',
    description: 'ばー',
    speakers: ['jacopen1', 'r_takaishi2'],
    onAir: true,
  },
  {
    id: 3,
    trackId: '1',
    vimeoId: '453963160',
    title: 'ほげほげほげ',
    description: 'ぴよぴよぴよ',
    speakers: ['jacopen1', 'r_takaishi2'],
  },
  {
    id: 4,
    trackId: '1',
    vimeoId: '453941938',
    title: 'ほげーーーーーー',
    description: 'ぴよーーーーーーー',
    speakers: ['jacopen1', 'r_takaishi2'],
  },
  {
    id: 5,
    trackId: '2',
    vimeoId: '453942665',
    title: 'こんぺこ',
    description: 'ぴよーーーーーーー',
    speakers: ['jacopen1', 'r_takaishi2'],
  },
  {
    id: 6,
    trackId: '2',
    vimeoId: '453943218',
    title: 'にゃっはろー',
    description: 'ぴよーーーーーーー',
    speakers: ['jacopen1', 'r_takaishi2'],
    onAir: true,
  },
  {
    id: 7,
    trackId: '2',
    vimeoId: '453939601',
    title: 'てすとてすと',
    description: 'ぴよーーーーーーー',
    speakers: ['jacopen1', 'r_takaishi2'],
  },
  {
    id: 8,
    trackId: '3',
    vimeoId: '453936668',
    title: 'あいうえおかきくけこ',
    description: 'ぴよーーーーーーー',
    speakers: ['jacopen1', 'r_takaishi2'],
    onAir: true,
  },
  {
    id: 9,
    trackId: '4',
    vimeoId: '453936358',
    title: 'ぷぎゃー',
    description: 'ぴよーーーーーーー',
    speakers: ['jacopen1', 'r_takaishi2'],
    onAir: true,
  },
  {
    id: 10,
    trackId: '5',
    vimeoId: '453935827',
    title: 'ぽぎゃー',
    description: 'ぴよーーーーーーー',
    speakers: ['jacopen1', 'r_takaishi2'],
    onAir: true,
  },
  {
    id: 11,
    trackId: '6',
    vimeoId: '450845161',
    title: 'ばー',
    description: 'ぴよーーーーーーー',
    speakers: ['jacopen1', 'r_takaishi2'],
    onAir: true,
  },
]
