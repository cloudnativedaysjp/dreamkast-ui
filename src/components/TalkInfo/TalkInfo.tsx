import React, { useEffect, useState, useCallback } from 'react'
import * as Styled from './styled'
import { Event, Talk, TrackApi, Configuration } from '../../client-axios'

type Props = {
  event?: Event
  selectedTalk?: Talk
  selectedTrackName?: string
  selectedTrackId?: number
}

export const TalkInfo: React.FC<Props> = ({
  event,
  selectedTalk,
  selectedTrackName,
  selectedTrackId,
}) => {
  const [timer, setTimer] = useState<number>()
  const [viewerCount, setViewerCount] = useState<string>()

  const twitterURL = (trackName?: string) => {
    const base = `http://twitter.com/share?url=https://event.cloudnativedays.jp/${event?.abbr}&related=@cloudnativedays&hashtags=${event?.abbr}`
    if (!trackName) return base
    return base + '_' + trackName
  }

  const getViewerCount = useCallback(async () => {
    console.log('Start getViewerCount')
    const api = new TrackApi(
      new Configuration({ basePath: window.location.origin }),
    )
    if (selectedTrackId) {
      try {
        console.log('Trying request')
        const { data } = await api.apiV1TracksTrackIdViewerCountGet(
          selectedTrackId.toString(),
        )
        setViewerCount(data.viewer_count.toString())
      } catch {
        setViewerCount('-')
      }
    }
  }, [viewerCount, selectedTrackId])

  useEffect(() => {
    getViewerCount()
    clearInterval(timer)
    setTimer(
      window.setInterval(() => {
        getViewerCount()
      }, 60 * 1000),
    )
  }, [selectedTrackId])

  return (
    <Styled.OuterContainer>
      <Styled.Container>
        {selectedTalk?.onAir && (
          <Styled.Live>LIVE üë• {viewerCount}</Styled.Live>
        )}
        <Styled.Title>{selectedTalk?.title}</Styled.Title>
        <Styled.SpeakerContainer>
          <Styled.Speaker>
            {selectedTalk?.speakers
              .map((speaker) => {
                return speaker.name
              })
              .join(' / ')}
          </Styled.Speaker>
          <div style={{ paddingRight: '20px' }} />
          {selectedTalk?.documentUrl && (
            <Styled.DocsLink href={selectedTalk?.documentUrl} target="_blank">
              ÁôªÂ£áË≥áÊñô„ÅØ„Åì„Å°„Çâ
            </Styled.DocsLink>
          )}
        </Styled.SpeakerContainer>
        <Styled.Content>{selectedTalk?.abstract}</Styled.Content>
        <Styled.SocialHeader>
          <Styled.TalkIcon src={`/${event?.abbr}/ui/talk_icon.png`} />
          ‰∏ÄÁ∑í„Å´Áõõ„Çä‰∏ä„Åå„Çç„ÅÜ
        </Styled.SocialHeader>
        <Styled.ButtonContainer>
          <Styled.ButtonLink
            href={twitterURL(selectedTrackName)}
            target="_blank"
          >
            <Styled.TweetButton>
              <Styled.TwitterImg src={`/${event?.abbr}/ui/twitter_logo.png`} />
              {`tweet #${event?.abbr}_${selectedTrackName}`}
            </Styled.TweetButton>
          </Styled.ButtonLink>
        </Styled.ButtonContainer>
      </Styled.Container>
    </Styled.OuterContainer>
  )
}
