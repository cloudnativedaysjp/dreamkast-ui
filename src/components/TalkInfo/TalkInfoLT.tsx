import React from 'react'
import * as Styled from './styled'

type Props = {
  eventAbbr: string
  title: string
  content: string
}

export const TalkInfoLT: React.FC<Props> = ({ eventAbbr, title, content }) => {
  const xURL = `http://x.com/share?url=https://event.cloudnativedays.jp/${eventAbbr}&related=@cloudnativedays&hashtags=${eventAbbr}`
  return (
    <Styled.Container>
      <Styled.Title>{title}</Styled.Title>
      <Styled.Content>{content}</Styled.Content>
      <Styled.SocialHeader>
        <Styled.TalkIcon src={`/${eventAbbr}/ui/talk_icon.png`} />
        一緒に盛り上がろう
      </Styled.SocialHeader>
      <Styled.ButtonContainer>
        <Styled.ButtonLink href={xURL} target="_blank">
          <Styled.XPostButton>
            <Styled.XImg src={`/${eventAbbr}/ui/x_logo.png`} />
            {`ポスト #${eventAbbr}`}
          </Styled.XPostButton>
        </Styled.ButtonLink>
      </Styled.ButtonContainer>
    </Styled.Container>
  )
}
