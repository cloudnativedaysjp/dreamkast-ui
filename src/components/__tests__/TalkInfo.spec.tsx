import React from 'react'
import renderer from 'react-test-renderer'
import { Talks } from '../../utils/talk-data'
import TalkInfo from '../TalkInfo'

test('TalkInfo', () => {
  const component = renderer.create(<TalkInfo selectedTalk={Talks[0]} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
