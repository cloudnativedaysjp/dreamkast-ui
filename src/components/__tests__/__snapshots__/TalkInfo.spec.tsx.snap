// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TalkInfo 1`] = `
<section
  className="talk-info"
>
  <h3>
    ほげふが
  </h3>
  <h4>
    jacopen1 / r_takaishi2
  </h4>
  <div>
    ぴよぴよ
  </div>
</section>
`;
